services/src/main/java/io/druid/cli/PullDependencies.java

```diff
@@ -53,23 +96,265 @@
+    unless debug logging is turned on.  "Disabling" it, however, is kinda bass-ackwards.  We copy out a reference
+    to the current System.out, and set System.out to a noop output stream.  Then after DefaultTeslaAether has pulled
+    The reference we swap things back.
+
+    This has implications for other things that are running in parallel to this.  Namely, if anything else also grabs
+    a reference to System.out or tries to log to it while we have things adjusted like this, then they will also log
+    to nothingness.  Fortunately, the code that calls this is single-threaded and shouldn't hopefully be running
+    alongside anything else that's grabbing System.out.  But who knows.
+    */
+
+    List<String> remoteUriList = extensionsConfig.getRemoteRepositories();
+
+    List<Repository> remoteRepositories = Lists.newArrayList();
+    for (String uri : remoteUriList) {
+      try {
+        URI u = new URI(uri);
+        Repository r = new Repository(uri);
+
+        if (u.getUserInfo() != null) {
+          String[] auth = u.getUserInfo().split(":", 2);
+          if (auth.length == 2) {
+            r.setUsername(auth[0]);
+            r.setPassword(auth[1]);
+          } else {
+            log.warn(
+                "Invalid credentials in repository URI, expecting [<user>:<password>], got [%s] for [%s]",
+                u.getUserInfo(),
+                uri
+            );
+          }
+        }
+        remoteRepositories.add(r);
+      }
+      catch (URISyntaxException e) {
+        throw Throwables.propagate(e);
+      }
+    }
+
+    if (log.isTraceEnabled() || log.isDebugEnabled()) {
+      return new DefaultTeslaAether(
+          extensionsConfig.getLocalRepository(),
+          remoteRepositories.toArray(new Repository[remoteRepositories.size()])
+      );
+    }
+
+    PrintStream oldOut = System.out;
+    try {
+      System.setOut(
+          new PrintStream(
+              new OutputStream()
+              {
+                @Override
+                public void write(int b) throws IOException
+                {
+
+                }
+
+                @Override
+                public void write(byte[] b) throws IOException
+                {
+
+                }
+
+                @Override
+                public void write(byte[] b, int off, int len) throws IOException
+                {
+
+                }
+              }
+              , false, StringUtils.UTF8_STRING
+          )
+      );
+      return new DefaultTeslaAether(
+          extensionsConfig.getLocalRepository(),
+          remoteRepositories.toArray(new Repository[remoteRepositories.size()])
+      );
+    }
+    catch (UnsupportedEncodingException e) {
+      // should never happen
+      throw new IllegalStateException(e);
+    }
+    finally {
+      System.setOut(oldOut);
+    }
+  }
+
+  private void createRootExtensionsDirectory(File atLocation)
+  {
+    if (!atLocation.mkdirs()) {
+      throw new RuntimeException(
```

Comment:
For this one, I think it still makes more sense to throw a RuntimeException or its subclass IllegalArgumentException as you suggested.